hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NSSummaryEndpoint.java

```diff
@@ -0,0 +1,640 @@
+   * @return the entity type, unknown if path not found
+   */
+  @VisibleForTesting
+  public EntityType getEntityType(String[] names) throws IOException {
+    if (names.length == 0) {
+      return EntityType.UNKNOWN;
+    } else if (names.length == 1) { // volume level check
+      String volName = names[0];
+      if (!checkVolumeExistence(volName)) {
+        return EntityType.UNKNOWN;
+      }
+      return EntityType.VOLUME;
+    } else if (names.length == 2) { // bucket level check
+      String volName = names[0];
+      String bucketName = names[1];
+      if (!checkBucketExistence(volName, bucketName)) {
+        return EntityType.UNKNOWN;
+      }
+      return EntityType.BUCKET;
+    } else { // length > 3. check dir or key existence (FSO-enabled)
+      String volName = names[0];
+      String bucketName = names[1];
+      String keyName = getKeyName(names);
+      // check if either volume or bucket doesn't exist
+      if (!checkVolumeExistence(volName)
+              || !checkBucketExistence(volName, bucketName)) {
+        return EntityType.UNKNOWN;
+      }
+      long bucketObjectId = getBucketObjectId(names);
+      return determineKeyPath(volName, bucketName, keyName, bucketObjectId);
+    }
+  }
+
+  /**
+   * Given a existent path, get the bucket object ID.
+   * @param names valid path request
+   * @return bucket objectID
+   * @throws IOException
+   */
+  private long getBucketObjectId(String[] names) throws IOException {
+    String bucketKey = omMetadataManager.getBucketKey(names[0], names[1]);
+    OmBucketInfo bucketInfo = omMetadataManager
+            .getBucketTable().getSkipCache(bucketKey);
+    return bucketInfo.getObjectID();
+  }
+
+  /**
+   * Given a valid path request for a directory,
+   * return the directory object ID.
+   * @param names parsed path request in a list of names
+   * @return directory object ID
+   */
+  private long getDirObjectId(String[] names) throws IOException {
+    return getDirObjectId(names, names.length);
+  }
+
+  /**
+   * Given a valid path request and a cutoff length where should be iterated
+   * up to.
+   * return the directory object ID for the object at the cutoff length
+   * @param names parsed path request in a list of names
+   * @param cutoff cannot be larger than the names' length. If equals,
+   *               return the directory object id for the whole path
+   * @return directory object ID
+   */
+  private long getDirObjectId(String[] names, int cutoff) throws IOException {
+    long dirObjectId = getBucketObjectId(names);
+    String dirKey = null;
+    for (int i = 2; i < cutoff; ++i) {
+      dirKey = omMetadataManager.getOzonePathKey(dirObjectId, names[i]);
+      OmDirectoryInfo dirInfo =
+              omMetadataManager.getDirectoryTable().getSkipCache(dirKey);
+      dirObjectId = dirInfo.getObjectID();
+    }
+    return dirObjectId;
+  }
+
+  @VisibleForTesting
+  public static String[] parseRequestPath(String path) {
```

Comment:
Make the method package-private by removing 'public` modifier, as this is only used outside the class in the test `TestNSSummaryEndpoint`.
```suggestion
  static String[] parseRequestPath(String path) {
```